youtube_dl/extractor/loom.py

```diff
@@ -0,0 +1,202 @@
+        {
+            'url': 'https://www.loom.com/share/31b41727a5b24dacb6c1417a565b2ebf',
+            'md5': '8b94361aabff2075141dc60bd6d35453',
+            'info_dict': {
+                'id': '31b41727a5b24dacb6c1417a565b2ebf',
+                'ext': 'mp4',
+                'title': 'How to resize your camera bubble',
+                'uploader': 'Allie Hitchcock',
+                'upload_date': '20201007',
+                'timestamp': 1602089241
+            }
+        },
+        {
+            'url': 'https://www.loom.com/share/7e5168ec3b0744cab5e08a340cc7e086',
+            'md5': '47dd14aa1d8054c249b68ca57ad9963f',
+            'info_dict': {
+                'id': '7e5168ec3b0744cab5e08a340cc7e086',
+                'ext': 'mp4',
+                'title': 'How to flip your camera ',
+                'uploader': 'Matthew Flores',
+                'upload_date': '20200423',
+                'timestamp': 1587646164
+            }
+        },
+        {
+            'url': 'https://www.loom.com/share/6670e3eba3c84dc09ada8306c7138075',
+            'md5': 'bfad8181ed49d6252b10dfdeb46c535e',
+            'info_dict': {
+                'id': '6670e3eba3c84dc09ada8306c7138075',
+                'ext': 'mp4',
+                'title': 'How to record your first video on Loom',
+                'uploader': 'Allie Hitchcock',
+                'upload_date': '20201118',
+                'timestamp': 1605729404
+            }
+        }
+    ]
+
+    def _extract_video_info_json(self, webpage, video_id):
+        info = self._html_search_regex(
+            r'window.loomSSRVideo = (.+?);',
+            webpage,
+            'info')
+        return self._parse_json(info, 'json', js_to_json)
+
+    def _get_url_by_id_type(self, video_id, type):
+        request = compat_urllib_request.Request(
+            self._BASE_URL + 'api/campaigns/sessions/' + video_id + '/' + type,
+            {})
+        json_doc = self._download_json(request, video_id)
+        return (url_or_none(json_doc.get('url')), json_doc.get('part_credentials'))
+
+    def _get_m3u8_formats(self, url, video_id, credentials):
+        format_list = self._extract_m3u8_formats(url, video_id)
+        for item in format_list:
+            item['protocol'] = 'm3u8_native'
+            item['url'] += '?' + credentials
+            item['ext'] = 'mp4'
+            item['format_id'] = 'hls-' + str(item.get('height', 0))
+            item['extra_param_to_segment_url'] = credentials
+        return format_list
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        info = self._extract_video_info_json(webpage, video_id)
+
+        formats = []
+        for type in ['transcoded-url', 'raw-url']:
+            (url, part_credentials) = self._get_url_by_id_type(video_id, type)
+            ext = self._search_regex(
+                r'\.([a-zA-Z0-9]+)\?',
+                url, 'ext', default=None)
```

Comment:
Read coding conventions.

Reply:
For this part, I may need to extract the file extension from a url.

Would you prefer a relaxed regex `\.([^.?]+)\?`?

Or `HEAD [URL]` and extract the extension from `content-type` header with `mimetype2ext(mt)`?